docs/hugo/content/design/ADR-2024-02-Upstream-Deletion/_index.md

```diff
@@ -0,0 +1,52 @@
+
+However, we decided that this question still needed to be considered, as it's likely to come up again in the future and we need to have a plan in place for how to handle it.
+
+Fortunately, the `azure-rest-api-specs` is very tightly controlled and is used as the source of truth for code generation of client APIs across multiple languages (including Python, Java, C# and Go) so it's vanishingly unlikely for a resource version to be removed without warning. 
+
+### Scenario A: Deletion of preview version
+
+The most likely scenario is that an old preview version of a service is removed. Preview versions are only guaranteed to be supported for 90 days after creation, though in practice they are retained much much longer.
+
+> When releasing a new preview, the service team may completely retire any previous preview versions after giving customers at least 90 days to upgrade their code.
+  -- [Microsoft Azure REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md), retrieved 23 Feb 2024
+
+In this situation, the resource provider will stop accepting the old version of the resource, requiring all users to update to a newer api version, including users of ASO.
+
+ASO can support migration by including a later version of the resource in a new release.
+
+If the new ASO release is available before the deprecation/deletion date, users can update their clusters to the new version, and then update their resources to the new version at any time. As long as they complete the update prior to deletion, no downtime will occur.
+
+If the new ASO release is not available until after the depredcation/deletion date, users will need to export/recreate their resources as a part of the upgrade process.
+
+In either case, the removal of the old version from ASO will be a breaking change that must be documented on the breaking release, and should be documented on earlier releases if possible.
+
+### Scenario: Deletion of GA version
+
+The deletion of a GA version of a resource is much rarer event, and requires a much longer notice and deprecation period. 
+
+ASO should be aware of this change well in advance, and should be able to support migration by including a later version of the resource in a new release. We should also document the upcoming deprecation so that users know they need to migrate.
+
+### Scenario: Deletion of an entire resource
+
+Even rarer, this can happen when a service is retired or replaced. ASO should be aware of this change well in advance. Where possible, we should provide a migration path to the new service, and document the upcoming deprecation so that users know they need to migrate.
+
+## Decision
+
+Deletion of an upstream resource inevitably results in that resource being removed from ASO as we don't retain sufficient information to regenerate the resource. This is a breaking change.
```

Comment:
Not only do we not retain sufficient information to regenerate the resource, more importantly the actual resource will not work so even if we could generate it, it would not be useful.

IMO better to focus on that as that's the salient point, and mention the inability to generate as a second aspect that is less critical (since it won't work even if generated).

Reply:
Good point, my focus was too narrow.